var/spack/repos/builtin/packages/py-rich/package.py

```diff
@@ -28,7 +29,9 @@ class PyRich(PythonPackage):
     version('9.4.0', sha256='bde23a1761373fed2802502ff98292c5d735a5389ed96f4fe1be5fb4c2cde8ea')
 
     depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('python@3.6.2:', type=('build', 'run'), when='@10.14.0:')
     depends_on('py-setuptools', type='build')
+    depends_on('py-typing-extensions@3.7.4:4', type=('build', 'run'), when='@10.14.0: ^python@:3.7')
     depends_on('py-typing-extensions@3.7.4:3', type=('build', 'run'))
```

Comment:
```suggestion
    depends_on('py-typing-extensions@3.7.4:3', type=('build', 'run'), when='@:10.0')
```
Otherwise this constraint will be applied to all versions